### PR TITLE
add a new api to remove all the cert and pkey from ssl_ctx

### LINKED ISSUE #13216: Can i pull code adding a new api to remove all the cert and pkey from ssl_ctx?
I just want a api can help me do the remove operation when i set much cert to the ssl_ctx and i don't want their anymore.

### PR DESCRIPTION
**Pull request by stardrift1**
CLA: trivial
Fixes #13216

### CODE REVIEW

File: ssl/ssl_lib.c [hunk 1]
```diff
@@ -4132,6 +4132,26 @@ EVP_PKEY *SSL_CTX_get0_privatekey(const SSL_CTX *ctx)
         return NULL;
 }
 
+int SSL_CTX_remove_cert_and_key(SSL_CTX *ctx)
+{
+    if(!ctx) {
```
Comment (paulidale) [inline ssl/ssl_lib.c:4137]: Style: space between the `if` and the `(`.

Drop `{` and `}` around single statement bodies like this one.

File: ssl/ssl_lib.c [hunk 2]
```diff
@@ -4132,6 +4132,26 @@ EVP_PKEY *SSL_CTX_get0_privatekey(const SSL_CTX *ctx)
...
+        return 0;
+    }
+
+    if(!CRYPTO_THREAD_write_lock(ctx->cert->lock)) {
```
Comment (paulidale) [inline ssl/ssl_lib.c:4141]: Style: space between the `if` and the `(`.

Drop `{` and `}` around single statement bodies like this one.

File: include/openssl/ssl.h.in [hunk 3]
```diff
@@ -2029,6 +2029,7 @@ struct evp_pkey_st *SSL_get_privatekey(const SSL *ssl);
 
 __owur X509 *SSL_CTX_get0_certificate(const SSL_CTX *ctx);
 __owur EVP_PKEY *SSL_CTX_get0_privatekey(const SSL_CTX *ctx);
+__owur int SSL_CTX_remove_cert_and_key(SSL_CTX *ctx);
```
Comment (paulidale) [inline include/openssl/ssl.h.in:2032]: You are adding a new public API.  You  need to do `make update` to regenerate some things.

File: ssl/ssl_lib.c [hunk 4]
```diff
@@ -4132,6 +4132,26 @@ EVP_PKEY *SSL_CTX_get0_privatekey(const SSL_CTX *ctx)
         return NULL;
 }
 
+int SSL_CTX_remove_cert_and_key(SSL_CTX *ctx)
```
Comment (paulidale) [inline ssl/ssl_lib.c:4135]: I think I'd prefer `SSL_CTX_remove_certificate_and_key` for the function name, but it's not a blocking point.

**Review by paulidale — COMMENTED** [review]
This is most definitely **not** a trivial pull request.  We **need** an ICLA before this can be accepted.

**Comment by richsalz** [comment]
Use explicit checks against NULL not things like `!ctx`

File: ssl/ssl_lib.c (hunk 4 above)
Comment (kaduk) [reply ssl/ssl_lib.c:4135]: This looks analogous to the existing `SSL_CTX_use_cert_and_key()`.

**Pull request closed without merging**